src/resolve/resolved-conf.c

```diff
@@ -385,6 +437,120 @@ int config_parse_dnssd_txt(const char *unit, const char *filename, unsigned line
         return 0;
 }
 
+int config_parse_dns_stub_listener_extra(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_free_ DNSStubListenerExtra *udp = NULL, *tcp = NULL;
+        _cleanup_free_ char *word = NULL, *s = NULL;
+        Manager *m = userdata;
+        bool both = false;
+        const char *p, *q;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                m->dns_extra_stub_listeners = ordered_hashmap_free(m->dns_extra_stub_listeners);
+                return 0;
+        }
+
+        p = rvalue;
+        r = extract_first_word(&p, &word, ":", 0);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r <= 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Invalid DnsStubListenExtra='%s, ignoring assignment", rvalue);
+                return 0;
+        }
+
+        if (!STR_IN_SET(word, "tcp", "udp"))
+                both = true;
+
+        if (streq(word, "tcp") || both) {
+                r = dns_stub_extra_new(&tcp);
+                if (r < 0)
+                        return log_oom();
+
+                tcp->mode = DNS_STUB_LISTENER_TCP;
+        }
+
+        if (streq(word, "udp") || both) {
+                r = dns_stub_extra_new(&udp);
+                if (r < 0)
+                        return log_oom();
+
+                udp->mode = DNS_STUB_LISTENER_UDP;
+        }
+
+        q = strrchr(rvalue, ':');
+        if (!q) {
+                s = strjoin(rvalue, ":53");
```

Comment:
The IP address format is like `[2001:db8:3c4d:15::1a2f:1a2b]:2201` . I think you are saying about if the format is like `2001:db8:3c4d:15::1a2f:1a2b` ?

Reply:
Right. If only IPv6 address is specified, e.g. `DNSStubListenerExtra=2001:db8:3c4d:15::1a2f:1a2b`, then `socket_address_parse()` fails.
Maybe, we should introduce `in_addr_port_from_string_auto()` and use it here, instead of using `socket_address_parse()`.

Reply:
Yes we need here

Reply:
introduced `socket_addr_port_from_string_auto` which required for such use cases.